src/main/java/com/github/os72/protocjar/maven/ProtocJarMojo.java

```diff
@@ -243,11 +243,52 @@ public void execute() throws MojoExecutionException {
 			if (target.outputDirectorySuffix != null) {
 				target.outputDirectory = new File(target.outputDirectory, target.outputDirectorySuffix);
 			}
+            // Copy files to a temporary directory.
+            target.outputDirectoryFinal = target.outputDirectory;
+            target.outputDirectory = new File(target.outputDirectory.getParent(), target.outputDirectory.getName() + "-tmp");
+
 		}
 		
 		performProtoCompilation();
 	}
 
+    private void copyAndUpdateOutputTarget(OutputTarget outputTarget) throws MojoExecutionException {
+        File src = outputTarget.outputDirectory;
+        File dst = outputTarget.outputDirectoryFinal;
+
+        dst.mkdirs();
+
+        // List files in dst and copy them to target.
+        Collection<File> files = org.apache.commons.io.FileUtils.listFiles(src, null, true);
```

Comment:
Isn't this the code I had written a while back?  What lines did you add?

Actually, there's a bug in my implementation.  I should list files in the temporary directory, copy the new/changed ones to target, and also remove files in target that aren't in the temporary directory (I forgot this step).

Could you add that as well if you can?  And then, it would be good to push this to our maven repo, run pmr, and see that the build goes through.

In this plugin directory:
```
mvn -DaltDeploymentRepository=central::default::http://artifactory.tetrationanalytics.com:8081/artifactory/plugins-snapshot-local deploy                     
```

Then, change pom.xml in our projects to point to the new maven protobuf compiler, and then run pmr.

thanks,